### PR TITLE
Add `reporter` option

### PR DESCRIPTION
**Pull request by damccorm**
I currently am trying to add CI to ```sindresorhus/awesome``` using Azure Pipelines, and I would like to use a custom reporter so that I can display test results in the Pipelines test UI. This PR enables that by adding support for custom reporters.

PR here: https://github.com/sindresorhus/awesome/pull/1505

### CODE REVIEW

**Comment by damccorm** [comment]
@sindresorhus @transitive-bullshit could you take a look at this?

File: index.js [hunk 1]
```diff
@@ -69,7 +73,11 @@ lint.report = async options => {
 	process.exitCode = 1;
 
 	file.path = path.basename(file.path);
-	console.log(vfileReporterPretty([file]));
+	if (options.customReporter) {
```
Comment (sindresorhus) [inline index.js]: The option name can simply be named `reporter`.

File: index.js [hunk 2]
```diff
@@ -69,7 +73,11 @@ lint.report = async options => {
...
+		console.log(options.customReporter([file]));
+	} else {
+		console.log(vfileReporterPretty([file]));
+	}
```
Comment (sindresorhus) [inline index.js]: Can be simplified to:

```js
const reporter = options.reporter || vfileReporterPretty;
console.log(reporter([file]));
```

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
 	"name": "awesome-lint",
-	"version": "0.8.0",
+	"version": "0.9.0",
```
Comment (sindresorhus) [inline package.json]: Don't bump version in a pull request. It's up to the maintainers when, how, and what version to bump :)

**Comment by sindresorhus** [comment]
Yeah, makes sense to have this. The option must be documented in the readme and needs a test in https://github.com/sindresorhus/awesome-lint/blob/master/test/cli.js

**Comment by sindresorhus** [comment]
It should be exposed as a CLI flag too.

File: package.json (hunk 3 above)
Comment (damccorm) [reply package.json]: Ah, my apologies, that makes sense.

**Comment by damccorm** [comment]
@sindresorhus I went ahead and responded to all feedback, let me know if you think there's any additional work to do before merging.

**Comment by sindresorhus** [comment]
Looks good :)

**Pull request merged**